mixer/adapter/rbac/rbacStore.go

```diff
@@ -304,8 +424,27 @@ func valueMatch(a interface{}, b string) bool {
 	return false
 }
 
-// Check if all properties defined for the subject are satisfied by the properties from the request.
-func checkSubject(properties map[string]interface{}, subject map[string]string) bool {
+// checkSubjectsMatch returns true if rule subjects and instance subjects are matched. Otherwise
+// returns false.
+func checkSubjectsMatch(subject *rbacproto.Subject, instance *authorization.Instance) bool {
+	if subject.GetUser() != "" && !stringMatch(instance.Subject.User, []string{subject.GetUser()}) {
+		return false
+	}
+
+	if subject.GetGroup() != "" && !stringMatch(instance.Subject.Groups, []string{subject.GetGroup()}) {
+		return false
+	}
+
+	if len(subject.GetProperties()) > 0 && !checkSubjectProperties(instance.Subject.Properties, subject.GetProperties()) {
+		return false
+	}
+
+	return true
```

Comment:
I think we probably should return false if `subject.GetUser() == "" && subject.GetGroup() == "" && len(subject.GetProperties()) == 0`?
Usually one of the user/group/properties should be non-empty, but if they are all unset, it's likely a bad subject and the semantics is nobady is granted for the permission.

Reply:
Good point. If no flag is provide, then we should return empty output
```
if instance.Subject.User == "" && instance.Subject.Groups == "" && len(instance.Subject.Properties) == 0 {
  return false
}
```